ambari-server/src/main/java/org/apache/ambari/server/topology/addservice/AddServiceInfo_Builder.java

```diff
@@ -0,0 +1,517 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ambari.server.topology.addservice;
```

Comment:
Given this file is generated by a framework that is not part of the project, can the generated builder be simplified to ease manual maintenance. E.g. probably unused methods such as mergeFrom() stripped, toString() replaced to a simpler implementation such as Objects.toString(...), etc.

Reply:
Also, given that AddServiceInfo will only be used internally and on a single thread, wouldn't be easier to use a simple mutable object rather than converting back and forth between builder and value object?

I generally prefer immutable value objects however and my opinion would be different if we had such a framework standardized with the project, but I am reluctant to maintain generated sources by hand unless there is a very strong reason (e.g. object is shared between independently developed classes, different threads)

Other people maintaining this file will probably not know the tool and instead of regenerating the file on any change they would make hand-made modifications which I think is error-prone.